sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/ApproximatePercentile.scala

```diff
@@ -238,12 +238,6 @@ object ApproximatePercentile {
       summaries = summaries.insert(value)
       // The result of QuantileSummaries.insert is un-compressed
       isCompressed = false
-
-      // Currently, QuantileSummaries ignores the construction parameter compressThresHold,
-      // which may cause QuantileSummaries to occupy unbounded memory. We have to hack around here
-      // to make sure QuantileSummaries doesn't occupy infinite memory.
-      // TODO: Figure out why QuantileSummaries ignores construction parameter compressThresHold
-      if (summaries.sampled.length >= compressThresHoldBufferLength) compress()
```

Comment:
I tested if this change doesn't cause `compress()` to not be called at all, and memory consumption to go ubounded, but it appears to be working good - the mem usage through jmap -histo:live when running `sql("select approx_percentile(id, array(0.1)) from range(10000000000L)").collect()` remains stable.
The compress() is being called from `QuantileSummaries.insert()`, so it seems that the above TODO got resolved at some point.

Reply:
Yes, the TODO was resolved in SPARK-17439. I thought I clearly stated it in the description, but if this is not the case or you have any suggestion about how to improve the description, I am happy to improve it.

Reply:
Sorry, it's my fault of not reading the description attentively :-).

Reply:
no problem at all, thanks for checking this :) I addressed you comment on the test. Any more comments?